satpy/writers/cf_writer.py

```diff
@@ -610,10 +610,12 @@ def save_datasets(self, datasets, filename=None, groups=None, header_attrs=None,
             if flatten_attrs:
                 header_attrs = flatten_dict(header_attrs)
             root.attrs = encode_attrs_nc(header_attrs)
-        root.attrs['history'] = 'Created by pytroll/satpy on {}'.format(datetime.utcnow())
+        if 'history' not in root.attrs:
+            root.attrs['history'] = 'Created by pytroll/satpy on {}'.format(datetime.utcnow())
```

Comment:
This is one case where I wonder if appending is the right thing to do. The history is meant to be a history of how the file was created, right? So maybe appending something to the history would be better than overwriting it? I don't have strong feelings or experience with this though.

Reply:
I agree with @djhoese . Here is the relevant snippet from the CF conventions document:
```
history

    Provides an audit trail for modifications to the original data. Well-behaved generic netCDF filters will automatically append their name and the parameters with which they were invoked to the global history attribute of an input netCDF file. We recommend that each line begin with a timestamp indicating the date and time of day that the program was executed.
```

Reply:
OK. But you agree that if the user provide a history in `header_attrs` this should be appended to the hard coded _created by ..._ statement. But we rely on the user to provide this `history` attribute in the recommended format.